L1Trigger/CSCTriggerPrimitives/src/CSCGEMMotherboard.h

```diff
@@ -0,0 +1,450 @@
+{
+  matches<T> result, p1, p2;
+  p1 = matchingPads<T>(d1, part);
+  p2 = matchingPads<T>(d2, part);
+  result.reserve(p1.size() + p2.size());
+  result.insert(std::end(result), std::begin(p1), std::end(p1));
+  result.insert(std::end(result), std::begin(p2), std::end(p2));
+  return result;
+}
+
+template <class T>
+matches<T> CSCGEMMotherboard::matchingPads(const CSCCLCTDigi& clct1, const CSCALCTDigi& alct1, enum CSCPart part)
+{
+  matches<T> padsClct(matchingPads<T>(clct1, part));
+  matches<T> padsAlct(matchingPads<T>(alct1, part));
+  return intersection(padsClct, padsAlct);
+}
+
+template <class T>
+matches<T> CSCGEMMotherboard::matchingPads(const CSCCLCTDigi& clct1, const CSCCLCTDigi& clct2,
+					   const CSCALCTDigi& alct1, const CSCALCTDigi& alct2,
+					   enum CSCPart part)
+{
+  matches<T> padsClct(matchingPads<CSCCLCTDigi,T>(clct1, clct2, part));
+  matches<T> padsAlct(matchingPads<CSCALCTDigi,T>(alct1, alct2, part));
+  return intersection(padsClct, padsAlct);
+}
+
+template <class S>  
+void CSCGEMMotherboard::printGEMTriggerPads(int bx_start, int bx_stop, enum CSCPart part)
+{
+  bool iscopad = false;
+  // get the generic LUT  
+  boost::variant<GEMPadDigiIdsBX, GEMCoPadDigiIdsBX> variant_lut;
+  if (std::is_same<S, GEMPadDigi>::value) variant_lut = pads_;
+  else {
+    variant_lut = coPads_;
+    iscopad = true;
+  }
+  const auto& actual_lut = boost::get<std::map<int, matches<S> > >(variant_lut);
+
+  // pads or copads?
+  const bool hasPads(actual_lut.size()!=0);
+  
+  LogTrace("CSCGEMMotherboard") << "------------------------------------------------------------------------" << std::endl;
+  bool first = true;
+  for (int bx = bx_start; bx <= bx_stop; bx++) {
+    std::vector<std::pair<unsigned int, S> > in_pads = actual_lut[bx];
+    if (first) {
+      if (!iscopad) LogTrace("CSCGEMMotherboard") << "* GEM trigger pads: " << std::endl;
+      else          LogTrace("CSCGEMMotherboard") << "* GEM trigger coincidence pads: " << std::endl;
+    }
+    first = false;
+    if (!iscopad) LogTrace("CSCGEMMotherboard") << "N(pads) BX " << bx << " : " << in_pads.size() << std::endl;
+    else          LogTrace("CSCGEMMotherboard") << "N(copads) BX " << bx << " : " << in_pads.size() << std::endl;
+    if (hasPads){
+      for (const auto& pad : in_pads){ 
+	LogTrace("CSCGEMMotherboard") << "\tdetId " << GEMDetId(pad.first) << ", pad = " << pad.second;
+	auto roll_id(GEMDetId(pad.first));
+        if (part==CSCPart::ME11 and isPadInOverlap(GEMDetId(roll_id).roll())) LogTrace("CSCGEMMotherboard") << " (in overlap)" << std::endl;
+        else LogTrace("CSCGEMMotherboard") << std::endl;
+      }
+    }
+    else
+      break;
+  }
+}
+
+template <class S>  
+unsigned int CSCGEMMotherboard::findQualityGEM(const CSCALCTDigi& aLCT, const CSCCLCTDigi& cLCT)
+{
+  /*
+    Same LCT quality definition as standard LCTs
+    a4 and c4 take GEMs into account!!!
+  */
+  
+  unsigned int quality = 0;
```

Comment:
This function assigns the quality for an LCT. The LCT quality is an int and has values 0 to 15. Changing it to an enum would only cause confusion as those numbers have been used for years.